File: services/19-aspnet-core/App/Controllers/HelloController.cs
using Microsoft.AspNetCore.Mvc;

namespace App.Controllers;

[ApiController]
[Route("")]
public class HelloController : ControllerBase
{
    [HttpGet("/")] public IActionResult Hello() =>
        Ok(new { message = "Hello from ASP.NET Core 9", framework = "19-aspnet-core", version = "1.0.0" });

    [HttpGet("/health")] public IActionResult Health() =>
        Ok(new { status = "ok", version = "1.0.0" });

    [HttpGet("/health/live")] public IActionResult Liveness() => Ok(new { status = "ok" });
    [HttpGet("/health/ready")] public IActionResult Readiness() => Ok(new { status = "ok" });
}

File: services/17-micronaut/src/main/java/com/example/HelloController.java
package com.example;

import io.micronaut.http.annotation.*;
import java.util.Map;

@Controller
public class HelloController {

    @Get("/")
    public Map<String, String> hello() {
        return Map.of("message", "Hello from Micronaut 5.0", "framework", "17-micronaut", "version", "1.0.0");
    }

    @Get("/health")
    public Map<String, String> health() {
        return Map.of("status", "ok", "version", "1.0.0");
    }

    @Get("/health/live")
    public Map<String, String> liveness() { return Map.of("status", "ok"); }

    @Get("/health/ready")
    public Map<String, String> readiness() { return Map.of("status", "ok"); }
}
